crates/bitwarden-crypto/src/store/mod.rs

```diff
@@ -259,6 +259,53 @@ impl<Ids: KeyIds> KeyStore<Ids> {
         res
     }
 
+    /// Decrypt a list of items using this key store, returning a tuple of successful and failed
+    /// items.
+    ///
+    /// # Arguments
+    /// * `data` - The list of items to decrypt.
+    ///
+    /// # Returns
+    /// A tuple containing two vectors: the first vector contains the successfully decrypted items,
+    /// and the second vector contains the items that failed to decrypt along with their error
+    /// views.
+    pub fn decrypt_list_with_failures<
+        'a,
+        Key: KeyId,
+        Data: Decryptable<Ids, Key, Output> + IdentifyKey<Key> + Send + Sync + 'a,
+        Output: Send + Sync,
+        ErrorView: From<&'a Data> + Send + Sync,
+    >(
+        &self,
+        data: &'a [Data],
+    ) -> (Vec<Output>, Vec<ErrorView>) {
+        let results: (Vec<_>, Vec<_>) = data
+            .par_chunks(batch_chunk_size(data.len()))
+            .map(|chunk| {
+                let mut ctx = self.context();
+
+                let mut result = Vec::with_capacity(data.len());
+
+                for item in chunk {
+                    let key = item.key_identifier();
+                    result.push(
+                        item.decrypt(&mut ctx, key)
+                            .map_err(|_| ErrorView::from(item)),
+                    );
+                    ctx.clear_local();
+                }
+
+                result
```

Comment:
Seems we can make this a little bit more concise: 
```suggestion
                let mut ctx = self.context();

                chunk
                    .iter()
                    .map(|item| {
                        let result = item
                            .decrypt(&mut ctx, item.key_identifier())
                            .map_err(|_| ErrorView::from(item));
                        ctx.clear_local();
                        result
                    })
                    .collect::<Vec<_>>()
```

Reply:
Yeah, it's a micro optimization really, but was easy to do so figured why not. In this example, doing chunk.iter().map().collect() will also apply the same optimization internally, so we can do it as well (The FromIterator of Vec is smart enough to use the SizeHint trait to preallocate the correct amount)

Reply:
Nice, thanks for the small cleanup!